bookkeeper-server/src/main/java/org/apache/bookkeeper/replication/Auditor.java

```diff
@@ -345,21 +348,73 @@ public void start() {
+                executor.submit(bookieCheck);
+            } else {
+                LOG.info("Auditor periodic bookie checking enabled"
+                         + " 'auditorPeriodicBookieCheckInterval' {} seconds", bookieCheckInterval);
+                executor.scheduleAtFixedRate(bookieCheck, 0, bookieCheckInterval, TimeUnit.SECONDS);
+            }
+            
             long interval = conf.getAuditorPeriodicCheckInterval();
 
             if (interval > 0) {
-                LOG.info("Auditor periodic ledger checking enabled"
-                         + " 'auditorPeriodicCheckInterval' {} seconds", interval);
+                LOG.info("Auditor periodic ledger checking enabled" + " 'auditorPeriodicCheckInterval' {} seconds",
+                        interval);
+
+                long checkAllLedgersLastExecutedCTime = -1;
+                long durationSinceLastExecutionInSecs = 0;
+                long initialDelay = 0;
+                try {
+                    checkAllLedgersLastExecutedCTime = ledgerUnderreplicationManager.getCheckAllLedgersCTime();
+                } catch (UnavailableException ue) {
+                    LOG.error("Got UnavailableException while trying to get checkAllLedgersCTime", ue);
+                }
+                if (checkAllLedgersLastExecutedCTime == -1) {
+                    durationSinceLastExecutionInSecs = -1;
+                    initialDelay = 0;
+                } else {
+                    durationSinceLastExecutionInSecs = (System.currentTimeMillis() - checkAllLedgersLastExecutedCTime)
+                            / 1000;
+                    initialDelay = durationSinceLastExecutionInSecs > interval ? 0
+                            : (interval - durationSinceLastExecutionInSecs);
+                }
+                LOG.info(
+                        "checkAllLedgers scheduling info.  checkAllLedgersLastExecutedCTime: {} "
+                                + "durationSinceLastExecutionInSecs: {} initialDelay: {} interval: {}",
+                        checkAllLedgersLastExecutedCTime, durationSinceLastExecutionInSecs, initialDelay, interval);
+
                 executor.scheduleAtFixedRate(new Runnable() {
                         public void run() {
                             try {
                                 if (!ledgerUnderreplicationManager.isLedgerReplicationEnabled()) {
-                                    LOG.info("Ledger replication disabled, skipping");
+                                    LOG.info("Ledger replication disabled, skipping checkAllLedgers");
                                     return;
                                 }
 
                                 Stopwatch stopwatch = Stopwatch.createStarted();
+                                LOG.info("Starting checkAllLedgers");
                                 checkAllLedgers();
+                                LOG.info("Completed checkAllLedgers");
```

Comment:
Since you added log info; add more info if possible. At least the time it took and maybe how many ledgers it checked??

Reply:
anyhow added log for duration of execution